pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -797,32 +836,57 @@ func (n *NetworkPolicyController) deleteNetworkPolicy(old interface{}) {
 func (n *NetworkPolicyController) addService(obj interface{}) {
 	defer n.heartbeat("addService")
 	service := obj.(*v1.Service)
-	klog.V(2).Infof("Processing Service %s/%s ADD event", service.Namespace, service.Name)
+	klog.V(2).InfoS("Processing Service ADD event", "serviceNamespace", service.Namespace, "serviceName", service.Name)
 	// Find all internal Group keys which refers to this Service.
 	groupKeySet := n.filterInternalGroupsForService(service)
 	// Enqueue internal groups to its queue for group processing.
 	for group := range groupKeySet {
 		n.enqueueInternalGroup(group)
 	}
+
+	// If this Service doesn't have a selector, there won't be an Endpoints ADD event
+	// triggered by this Service ADD event. So find all ACNPs that are impacted by this
+	// Service and reprocess them here. Otherwise, an Endpoints ADD event will
+	// reprocess those ACNPs.
+	if service.Spec.Selector == nil {
+		affectedACNPsByService := n.filterACNPsByService(service.Namespace, service.Name)
+		for _, acnp := range affectedACNPsByService {
+			n.reprocessACNP(acnp)
+		}
+	}
```

Comment:
is this specifically for the case where an Endpoints resource is created before the corresponding Service (without selector) resource? It's the only case I can think of but it's not obvious from the comment.

Reply:
You are right. That's the case. Added in the comment.